web-app/src/components/UserModal/container.js

```diff
@@ -24,27 +24,31 @@ const UserModalContainer = Wrapped =>
       this.state = {
         userDetails: null,
         userHolidays: [],
+        totalHolidays: 0,
       };
       this.hasPermission = props.userDetails.employeeRoleId !== roles.STANDARD;
     }
 
     componentDidMount() {
       const { user } = this.props;
       if (isEmpty(user)) return null;
-     
-      if (this.hasPermission) {
-        getHolidays(user.employeeId)
-          .then(response => {
-            this.setState({ userHolidays: response.data });
-          })
-          .catch(error => {
-            swal(
-              'Error',
-              `Error getting user holiday details: ${error.message}`,
-              'error'
-            );
-          });
-      }
+
+      getHolidays(user.employeeId)
+        .then(response => {
+          if (response.data.length > 0) {
+            this.setState({
+              totalHolidays: response.data[0].employee.totalHolidays,
+            });
+          }
+          this.setState({ userHolidays: response.data });
```

Comment:
Rather than having (potentially) two sequential setStates, would it be worth having this as 
```
this.setState({
  userHolidays: response.data,
  totalHolidays: response.data.length > 0 ? response.data[0].employee.totalHolidays : 0,
})
```